app/components/upload_item_shared/index.tsx

```diff
@@ -0,0 +1,313 @@
+            shadowColor: '#000000',
+            shadowOffset: {
+                width: 0,
+                height: 2,
+            },
+            shadowOpacity: 0.08,
+            shadowRadius: 3,
+            elevation: 1,
+        },
+        imageOnlyThumbnail: {
+            width: UPLOAD_ITEM_DIMENSIONS.THUMBNAIL_SIZE,
+            height: UPLOAD_ITEM_DIMENSIONS.THUMBNAIL_SIZE,
+            borderRadius: 4,
+            overflow: 'hidden',
+            shadowColor: '#000000',
+            shadowOffset: {
+                width: 0,
+                height: 2,
+            },
+            shadowOpacity: 0.08,
+            shadowRadius: 3,
+            elevation: 1,
+        },
+        fileInfo: {
+            flex: 1,
+            justifyContent: 'center',
+            alignItems: 'flex-start',
+            minWidth: 0,
+        },
+        fileName: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200, 'SemiBold'),
+            marginBottom: 2,
+            width: '100%',
+        },
+        fileSize: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75, 'Regular'),
+            width: '100%',
+        },
+        progress: {
+            position: 'absolute',
+            left: 0,
+            right: 0,
+            top: 0,
+            bottom: 0,
+            backgroundColor: 'rgba(0, 0, 0, 0.1)',
+            borderRadius: 4,
+            justifyContent: 'flex-end',
+        },
+        progressContainer: {
+            paddingVertical: undefined,
+            position: undefined,
+            justifyContent: undefined,
+        },
+        errorBorder: {
+            borderColor: theme.errorTextColor,
+            borderWidth: 2,
+        },
+    };
+});
+
+export default function UploadItemShared({
+    file,
+    onPress,
+    onRetry,
+    loading = false,
+    progress = 0,
+    showRetryButton = false,
+    galleryStyles,
+    testID,
+    fullWidth = false,
+    isShareExtension = false,
+    forwardRef,
+    inViewPort = false,
+    hasError = false,
+}: UploadItemProps) {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+
+    const fileForCheck = {
+        name: file.name,
+        extension: file.extension,
+        mime_type: file.mime_type,
+    } as FileInfo;
+
+    const isImageFile = isImage(fileForCheck);
+
+    const fileDisplayComponent = () => {
+        if (isImageFile) {
+            if (isShareExtension) {
+                return (
+                    <View style={style.imageOnlyThumbnail}>
+                        <Image
+                            source={{uri: file.uri}}
+                            style={{
+                                width: UPLOAD_ITEM_DIMENSIONS.THUMBNAIL_SIZE,
+                                height: UPLOAD_ITEM_DIMENSIONS.THUMBNAIL_SIZE,
+                                borderRadius: 4,
+                            }}
```

Comment:
not sure why this is inline but later on you memoize some styles.. 

can you clarify why certain things are memoized and why others are not ?

Reply:
Double-checked. Memorised wherever needed.